### PR TITLE
fix: types for common functions

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
see #7731 

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Common.php [hunk 1]
```diff
@@ -590,7 +592,7 @@ function function_usable(string $functionName): bool
      *
      * @throws FileNotFoundException
      */
-    function helper($filenames)
+    function helper($filenames): void
```
Comment (neznaika0) [inline system/Common.php:595]: @kenjis said to add PHPDoc comments

Comment (paulbalandan) [reply system/Common.php:595]: If a BC break or not supported by current PHP version (like intersections and unions or complex callable signatures), this one is neither of those.

**Review by neznaika0 — APPROVED** [review]

File: system/Common.php [hunk 2]
```diff
@@ -474,6 +474,8 @@ function esc($data, string $context = 'html', ?string $encoding = null)
      * @param int $duration How long should the SSL header be set for? (in seconds)
      *                      Defaults to 1 year.
      *
+     * @return never|void
```
Comment (kenjis) [inline system/Common.php:477]: @paulbalandan I probably wrote it somewhere, but isn't it wrong type?

> never is a return-only type indicating the function does not terminate. This means that it either calls [exit()](https://www.php.net/manual/en/function.exit.php), throws an exception, or is an infinite loop. Therefore, **it cannot be part of a [union type](https://www.php.net/manual/en/language.types.type-system.php#language.types.type-system.composite.union) declaration**. Available as of PHP 8.1.0.
https://www.php.net/manual/en/language.types.never.php

File: system/Common.php (hunk 1 above)
Comment (kenjis) [reply system/Common.php:595]: Functions cannot be extended in PHP. So the change will not break user app code.

File: system/Common.php (hunk 2 above)
Comment (paulbalandan) [reply system/Common.php:477]: Honestly, I didn't know `never` cannot be used in unions. TIL. I'll change.

**Review by kenjis — APPROVED** [review]

File: system/Common.php (hunk 2 above)
Comment (kenjis) [reply system/Common.php]: No problem. I also wrote `@return false|never|string`.

**Comment by paulbalandan** [comment]
Rebased.

**Pull request merged**